src/api/ApiIndex.vue

```diff
@@ -5,7 +5,7 @@
 import { data as apiIndex, APIGroup } from './api.data'
 import { ref, computed } from 'vue'
 
-const query = ref('')
+const query = ref<string>('')
```

Comment:
I'm unclear what these changes to `ApiIndex.vue` and `api.data.ts` are. They seem to be unrelated to the typos.

Reply:
@skirtles-code yeah, not related to the typos. It's just basically adding typing to query `ref` and destructuring the parent array on line 74.

Reply:
I think I preferred the code for the template of `ApiIndex.vue` the way it was originally. There might be a small amount of duplication, but given the nested loops I think it's clearer to use, e.g. `section.text` rather than just `text`. It helps to differentiate `section.text` from `item.text`.

What is the motivation behind explicitly adding the type information?